addons/hw_posbox_homepage/static/src/app/components/dialog/UpdateDialog.js

```diff
@@ -25,34 +25,75 @@ export class UpdateDialog extends Component {
         this.state.initialization = [];
     }
 
-    async getUpgradeData() {
+    async getGitTreeInfo() {
         try {
             const data = await this.store.rpc({
-                url: "/hw_posbox_homepage/upgrade",
+                url: "/hw_posbox_homepage/git_tree_info",
             });
 
-            this.state.upgradeData = data;
-            this.state.commitHtml = markup(data.commit);
+            this.state.isUpToDate = data.isUpToDate;
+            this.state.lastCommitMessage = data.lastCommitMessage;
             this.state.initialization = false;
         } catch {
             console.warn("Error while fetching data");
         }
     }
 
-    async upgradeIotBox() {
-        console.warn("Not implemented yet");
+    async updateGitTree() {
+        this.state.waitRestart = true;
+        try {
+            const data = await this.store.rpc({
+                url: "/hw_posbox_homepage/update_git_tree",
+                method: "POST",
+            });
+            if (data.status === "success") {
+                this.state.isUpToDate = true;
+            }
+        } catch {
+            console.warn("Error while updating IoT Box.");
+        }
+    }
+
+    async loadIotHandlers() {
+        this.state.waitRestart = true;
+        try {
+            const data = await this.store.rpc({
+                url: "/hw_posbox_homepage/load_iot_handlers",
+            });
+
+            if (data.status === "success") {
+                this.state.waitRestart = false;
+            }
+        } catch {
+            console.warn("Error while saving data");
```

Comment:
why this warning is so generic ? (same for the one below)

Reply:
i just copy-pasted, but that's a good point